daemonset-preloaded.yaml

```diff
@@ -0,0 +1,83 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Yaml file to install a DaemonSet that contains a device plugin for
+# managing nvidia gpu devices. Expected to be deployed in Kubernetes
+# clusters with nvidia gpu devices. On each node, the DaemonSet detects
+# whether there are nvidia gpu devices, and if so, sets up the devices
+# in ready state and report them to Kubelet through the device plugin
+# interface. The device plugin simply sleeps if there is no nvidia
+# gpu device on the node.
+
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: nvidia-driver-installer
+  namespace: kube-system
+spec:
+  template:
+    metadata:
+      labels:
+        name: nvidia-driver-installer
+    spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: cloud.google.com/gke-accelerator
+                operator: Exists
+      hostNetwork: true
+      hostPID: true
+      volumes:
+      - name: dev
+        hostPath:
+          path: /dev
+      - name: nvidia-overlay
+        hostPath:
+          path: /home/kubernetes/bin/nvidia
+      - name: os-release
+        hostPath:
+          path: /etc/os-release
+      - name: sysrq
+        hostPath:
+          path: /proc/sysrq-trigger
+      initContainers:
+      - image: "cos-nvidia-installer:fixed"
+        imagePullPolicy: Never
+        command: ["/bin/sh", "-c"]
+        args: ["usr/bin/nvidia-installer.sh"]
+        name: nvidia-driver-installer
+        resources:
+          requests:
+            cpu: 0.15
```

Comment:
Nit: should we increase this based on what we measured in PR https://github.com/kubernetes/kubernetes/pull/53541 that installer sometimes can take up to 2 cores?

Reply:
I am not sure. That would in effect reduce the capacity of all GPU nodes by 2 cores.

Reply:
I also have similar concern. What I am worried about is installer may take much longer time when the node is short of cpu. Perhaps we can discuss this later outside this PR.

/lgtm